src/base/logging.hh

```diff
@@ -123,19 +126,39 @@ class Logger
      */
     [[noreturn]] void exit_helper() { exit(); ::abort(); }
 
+    void registerExtraLog(std::function<std::string()> cb)
+    {
+        extraLog.emplace_back(std::move(cb));
+    }
+
   protected:
     bool enabled;
 
+    std::string getFormattedExtraLog()
+    {
+        std::stringstream ss;
+        for (auto &cb : extraLog) {
+            std::string str = cb();
+            ss << str;
+            if (str.length() && str.back() != '\n' && str.back() != '\r')
+                ss << std::endl;
+        }
+        return ss.str();
+    }
+
     /** Generates the log message. By default it is sent to cerr. */
     virtual void
     log(const Loc &loc, std::string s)
     {
+        s += getFormattedExtraLog();
         std::cerr << loc.file << ":" << loc.line << ": " << s;
```

Comment:
Nit: I find this more elegant:

` std::cerr << loc.file << ":" << loc.line << ": " << s << getFormattedExtraLog();`

Reply:
Thanks for the suggestion! Fixed!